tools/distpartitioning/data_shuffle.py

```diff
@@ -273,29 +263,25 @@ def exchange_node_features(rank, world_size, node_feature_tids, ntype_gnid_map,
                 local_idx_partid = local_idx[cond]
 
                 if (gnids_per_partid.shape[0] == 0):
-                    node_feats_per_rank.append({feat_key : torch.empty((0,), dtype=torch.float)})
-                    global_nid_per_rank.append({feat_key : torch.empty((0,), dtype=torch.int64)})
+                    node_feats_per_rank.append(torch.empty((0,1), dtype=torch.float))
+                    global_nid_per_rank.append(np.empty((0,1), dtype=np.int64))
                 else:
-                    node_feats_per_rank.append({feat_key : node_feats[local_idx_partid]})
-                    global_nid_per_rank.append({feat_key : gnids_per_partid})
+                    if (len(list(node_feats[local_idx_partid].size())) == 1):
+                        node_feats_per_rank.append(node_feats[local_idx_partid])
+                    else:
+                        node_feats_per_rank.append(node_feats[local_idx_partid])
+                    global_nid_per_rank.append(torch.from_numpy(gnids_per_partid).type(torch.int64))
 
             #features (and global nids) per rank to be sent out are ready
             #for transmission, perform alltoallv here.
-            output_feat_list = alltoall_cpu_object_lst(rank, world_size, node_feats_per_rank)
-            output_feat_list[rank] = node_feats_per_rank[rank]
+            output_feat_list = alltoallv_cpu_data(rank, world_size, node_feats_per_rank, node_feats.dtype)
 
-            output_nid_list = alltoall_cpu_object_lst(rank, world_size, global_nid_per_rank)
-            output_nid_list[rank] = global_nid_per_rank[rank]
+            #output_nid_list = alltoall_cpu_object_lst(rank, world_size, global_nid_per_rank)
```

Comment:
remove.

Reply:
Removed the commented out code as suggested.